sdk/core/azure-core/src/main/java/com/azure/core/http/policy/DefaultRedirectStrategy.java

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.policy;
+
+import com.azure.core.http.HttpHeaders;
+import com.azure.core.http.HttpMethod;
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.logging.ClientLogger;
+
+import java.net.HttpURLConnection;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * A default implementation of {@link RedirectStrategy} that uses the provided maximum retry attempts,
+ * header name to look up redirect url value for, http methods and a known set of
+ * redirect status response codes (301, 302, 307, 308) to determine if request should be redirected.
```

Comment:
Since the status codes are customizable too, we shouldn't list out these 4 status codes in class javadoc (it should mentioned in the 1st and 2nd ctors' javadocs).

Reply:
It is not customizable in the default implementation strategy.